File: 백준/Silver/10828. 스택/스택.cc
#include <bits/stdc++.h>
using namespace std;

int N, num;
string s;

int main(void) {
    ios_base::sync_with_stdio(0); cin.tie(0); cout.tie(0);

    cin >> N;

    stack<int> st;
    while(N--) {
        cin >> s;
        
        if(s == "push") {
            cin >> num;
            st.push(num);
        }
        if(s == "pop") {
            if(st.empty()) cout << -1 << '\n';
            else {cout << st.top() << '\n'; st.pop();}
        }
        if(s == "top") {
            if(st.empty()) cout << -1 << '\n';
            else {cout << st.top() << '\n';}
        }
        if(s == "size") {
            cout << st.size() << '\n';
        }
        if(s == "empty") {
            cout << st.empty() << '\n';
        }   
    }
}

File: 백준/Silver/10828. 스택/스택.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

fun main(args: Array<String>) {

    val br = BufferedReader(InputStreamReader(System.`in`));
    val bw = BufferedWriter(OutputStreamWriter(System.out));

    val N = br.readLine().toInt();
    val stack = MutableList<Int>(0) {it};

    (1..N).forEach() {
        val op = br.readLine().split(" ");

        if(op[0]=="push") {
            stack.add(op[1].toInt());
        }
        if(op[0]=="pop") {
            if(stack.isNotEmpty()) {
                bw.write(stack.last().toString());
                stack.removeLast();
            }
            else bw.write("-1");
            bw.write("\n");
        }
        if(op[0]=="size") {
            bw.write(stack.size.toString());
            bw.write("\n");
        }
        if(op[0]=="empty") {
            if(stack.isEmpty()) bw.write("1\n");
            else bw.write("0\n");
        }
        if(op[0]=="top") {
            if(stack.isNotEmpty()) {
                bw.write(stack.last().toString());
            }
            else bw.write("-1");
            bw.write("\n");
        }
    }

    br.close();
    bw.flush();
    bw.close();

}